04.ls/ls.rb

```diff
@@ -33,4 +33,118 @@ def split_files(files)
   divided_files
 end
 
+FILE_TYPE = {
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+
+PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+def load_l_option(path, file_names)
+  stat_files = file_names.map { |file_name| File.lstat("#{path}/#{file_name}") }
+  path_to_stat = stat_files.map.with_index { |stat, idx| ["#{path}/#{file_names[idx]}", stat] }.to_h
+
+  puts "total #{stat_files.sum(&:blocks)}"
+  load_body(path_to_stat).each do |file_show_data|
+    puts format_file(file_show_data, stat_files)
+  end
+end
+
+def load_body(path_to_stat)
+  path_to_stat.map do |absolute_path, stat|
+    {
+      permission: to_permission(stat),
+      link_count: stat.nlink.to_s,
+      owner: Etc.getpwuid(stat.uid).name,
+      group: Etc.getgrgid(stat.gid).name,
+      byte_size: stat.size.to_s,
+      time_stamp: stat.mtime.strftime(to_timestamp(stat.mtime)),
+      file_name: to_file_name(stat, absolute_path)
+    }
+  end
+end
+
+def to_timestamp(mtime)
+  branch_sec = 15_552_000
```

Comment:
```suggestion
  half_year = 15_552_000
```

とか、`six_months`とか、1555200秒ってどれくらいの期間なの？というのがわかる変数名の方がベターかな〜と思います

Reply:
`half_year`とてもわかりやすいです！使わせていただきました🙇‍♂️